examples/container_retries.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: container-retries-
+spec:
+  entrypoint: container-retries
+  templates:
+  - name: container-retries
+    retries:
```

Comment:
should retries be part of a step instead? ex: i have a workflow with multiple steps and some of the steps use the same template but with different retries.

Reply:
I agree, there is a use-case for retries with steps too. However, a step can itself be a workflow with multiple steps and that makes the semantics harder to express (e.g. if a step has multiple containers and some of these completed but others didn't, should the step be retried?).

Expressing the retries at the level of leaf node is easier and unambiguous. Hence this implementation has taken that approach as the first step.

As the implementation matures, retries could be added to the steps as well.